### PR TITLE
Tk auth docs

### PR DESCRIPTION
**Pull request by gguillotte**
Revise and expand tk-auth predocs. Notable changes:

- Split `configuration.markdown` into separate files.
- Added deprecation notices where applicable.
- Added a `deprecated_settings.md` similar to the Puppet docs to consolidate deprecation references.
- Expanded explanations of `auth.conf` parameters.
- Moved several links to Puppet API documentation from GitHub to the docs site.

Still WIP in fixing links; filing PR to preserve work and start getting feedback.

### CODE REVIEW

**Comment by gguillotte** [comment]
I've fixed some more links, and I think this is ready for review. @kayayarai @nfagerlund @camlow325

**Comment by camlow325** [comment]
Haven't reviewed this yet, but it appears there are some conflicts between this PR and what's currently on the HEAD of the master branch.

**Comment by gguillotte** [comment]
If the conflicts are blocking review, I'll need help ASAP resolving them. I've already implemented the changes from the conflicting commits into this branch.

**Comment by camlow325** [comment]
@ahpook Would be good if you could review this if/when you'd have time.

File: documentation/compatibility_with_puppet_agent.markdown [hunk 1]
```diff
@@ -1,159 +1,130 @@
...
+method find
+allow /^$1\.uuid.*/
+
+# Default rule, should follow the more specific rules
+path ~ ^/catalog/([^/]+)$
+method find
+allow $1
+~~~
+
+To support both Puppet 3 and Puppet 4 agents when the `use-legacy-auth-conf` parameter in the `jruby-puppet` setting is false, modify the rules to follow the new HOCON `auth.conf` format and place the new rules in `/etc/puppetlabs/puppet/conf.d/auth.conf`:
```
Comment (camlow325) [inline documentation/compatibility_with_puppet_agent.markdown]: The path reference at the end here should be `/etc/puppetlabs/puppetserver/conf.d/auth.conf` (adding "server" in the middle).

File: documentation/config_file_auth.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+[`trapperkeeper-authorization`]: https://github.com/puppetlabs/trapperkeeper-authorization
+[new `auth.conf`]: ./conf_file_auth.html
+[Puppet `auth.conf`]: /puppet/latest/reference/config_file_auth.html
+[deprecated]: ./deprecated_settings.html
+[`puppetserver.conf`]: ./conf_file_puppetserver.html
+
+The `auth.conf` file contains rules for authorizing access to Puppet Server's HTTP API endpoints. For a broader overview of Puppet Server configuration, see the [configuration documentation](./configuration.html).
+
+> **Deprecation Note:** As of version 2.2, Puppet Server can use [`trapperkeeper-authorization`][] for authentication, which is configured by rules and settings in `auth.conf` using a new HOCON configuration file format, and the new `conf.d` location for Puppet Server configuration files. The legacy [Puppet `auth.conf`][] whitelist-based authorization method and configuration file format are [deprecated][] and will be removed in a future version of Puppet Server.
```
Comment (camlow325) [inline documentation/config_file_auth.md]: Would it be better to say "The legacy Puppet 'auth.conf' *and* whitelist-based authorization method..."?  It's basically two things that we're deprecating in favor of tk-authorization for Puppet Server:

- Puppet's auth.conf for requests made to the master service endpoints
- client-whitelist settings for requests made to the certificate_status(es) and puppet-admin endpoints

**Comment by gguillotte** [comment]
OK, closing in favor of tk-auth-docs-2/PR #732.

**Pull request closed without merging**